pars-fmt/src/simple.rs

```diff
@@ -213,22 +256,28 @@ impl<'a> FmtMatcher<'a> {
             current_sep += 1;
         }
 
-        for (_field, sep) in self.fmt_fields.iter() {
+        for (field, sep) in self.fmt_fields.iter() {
             let sep_string = &self.source[sep.clone()];
+            let field_str = &self.source[field.clone()];
+            let field_idx = self
+                .fields
+                .iter()
+                .position(|f| f == field_str)
+                .expect("all fields have been validated");
 
             if pos == source.len() {
                 return Err(MatchError::InputExhausted);
             }
             if sep.start == sep.end {
                 // take all remaining string
-                values.push(pos..source.len());
+                values[field_idx] = pos..source.len();
                 pos = source.len();
                 continue;
             }
 
             match &source[pos..].find(sep_string) {
                 Some(idx) => {
-                    values.push(pos..pos + idx);
+                    values[field_idx] = pos..pos + idx;
```

Comment:
Not something to change necessarily (`rustfmt` will probably complain), but `pos..pos + idx` is weird syntax.

Reply:
maybe idx is a bad name? but I've gotten used to this syntax, Rust's operator precedence is (to my experience) pretty unambiguous?